public/app/plugins/datasource/prometheus/querybuilder/shared/operationUtils.ts

```diff
@@ -279,15 +279,20 @@ function getAggregationExplainer(aggregationName: string, mode: 'by' | 'without'
 
 function getAggregationByRendererWithParameter(aggregation: string) {
   return function aggregationRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
-    function mapType(p: QueryBuilderOperationParamValue) {
-      if (typeof p === 'string') {
+    const restParamIndex = def.params.findIndex((param) => param.restParam);
+    const params = model.params.slice(0, restParamIndex);
+    const restParams = model.params.slice(restParamIndex);
+
+    function mapType(p: QueryBuilderOperationParamValue, type: string) {
```

Comment:
Just a nit: I think with the mapping happening now in https://github.com/grafana/grafana/pull/52226/files#diff-50c2bec8e6c458efa349a71e1c7c740cb896cbd2d48f8e6b2f452d532d20e16dR294, this function is not really needed anymore, and the mapping could be done directly in line 294.